tests/resource_test.js

```diff
@@ -0,0 +1,65 @@
+/// <reference path="./steps.d.ts" />
+Feature('Resource test');
+
+var FirstLogin = false;
+var loginCookies = [];
+const pdfName = `Dummy PDF ${Math.random() * 1000}`;
+const collectionName = `Dummy Collection ${Math.random() * 1000}`;
+
+Before((I) => {
+    I.login('admin', 'password');
+});
+
+Scenario('test upload a PDF file and verify the upload process', (I) => {
+    I.seeInCurrentUrl('#dashboard');
+    I.click('Library'); I.wait(3);
+    I.seeInCurrentUrl('#resources');
+
+    I.click('Add New Resource'); I.wait(3);
+    I.seeInCurrentUrl('#resource/add');
+    I.see('New Resources');
+
+    I.see('Collection');
+    I.click('Add New'); I.wait(3);
+    I.fillField('CollectionName', collectionName);
+    I.fillField('Description', 'It is a dummy collection');
+    I.fillField('Year', '2017');
+    I.wait(3);
+    I.click('Save'); I.wait(3);
+    I.seeInPopup('Collection saved successfully.');
+    I.acceptPopup(); I.wait(3);
+    I.wait(3);
+    
+    I.fillField('Title', pdfName);
+    I.fillField('author', 'John Dou');
+    I.fillField('Year', '2017');
+    I.fillField('Publisher', 'Oh Really?');
+    I.fillField('linkToLicense', 'http://license.lic')
+    
+    // TODO: Click 'Select an Option' with a context
+    I.see('Select an Option');
+    I.click('Select an Option'); I.wait(3);
+    I.checkOption('Arts');
+    I.seeCheckboxIsChecked('Arts');
+    
+    I.see('Select an Option');
+    I.click('Select an Option'); I.wait(3);
+    I.checkOption('Undergraduate');
+    I.seeCheckboxIsChecked('Undergraduate');
+    
+
+    I.see('Select an Option');
+    I.click('Select an Option'); I.wait(3);
+    I.checkOption(collectionName);
+    I.seeCheckboxIsChecked(collectionName);
+    
+    I.attachFile('_attachments', 'assets/dummy.pdf'); I.wait(3);
+
+    I.click('//*[@id="resourceform"]/table/tbody/tr[4]/td[1]/div'); I.wait('2');
```

Comment:
I don't like it when we have to refer to elements like this - we should be able to refer to a button by its name or ID ideally. 

@dogi Should we make an issue on open-learning-exchange each time we run into this kind of problem?